src/Optional/Plots.jl

```diff
@@ -0,0 +1,33 @@
+using .Plots
+
+preprocess(name, val::Plots.Plot, data) = push!(data, name=>val)
+summary_impl(name, value::Plots.Plot) = image_summary(name, value)
+
+function image_summary(name::AbstractString, img::Plots.Plot)
+    pb = PipeBuffer()
+    show(pb, MIME("image/png"), img)
+    imdata = pb.data
+
+    # De-serializing PNG by hand. Because we're crazy.
+    # I want to get the h,w from the PNG because I don't trust Plots.jl
```

Comment:
Please link to Plots.jl issue?

Reply:
No Plots issue, but:
 - `Plots.Plot.args[:size]` is considered private API, so it could change in the future. I was trying to avoid using it;
 - It seems to me that `Plots.Plot.args` does not store the colourspace anyway; 
 - `plot(rand(3), size=(300,200.5)` is valid syntax for `Plots.jl` but when rendered to PNG it will be an integer. 

I'd reasonably say that the PNG format will not change in the future, so even if the code is somewhat obscure I believe it won't break no matter what happens to plotting backends.

If there was a package to do this, maybe on which we already depend, I'd prefer to use it, but I'm not aware of any.

Reply:
We could add a function `plotsize(p::Plot) = p.attr[:size]` to Plots if you need it. Not sure about what you need in terms of colorspace.

Reply:
If you can guarantee it would be properly rounded, it may help.

Regarding colourspace, taken from https://www.w3.org/TR/PNG-Chunks.html (colorspace). Ignore the Allowed bit depth column
```
   Color    Allowed    Interpretation
   Type    Bit Depths
   
   0       1,2,4,8,16  Each pixel is a grayscale sample.
   
   2       8,16        Each pixel is an R,G,B triple.
   
   3       1,2,4,8     Each pixel is a palette index;
                       a PLTE chunk must appear.
   
   4       8,16        Each pixel is a grayscale sample,
                       followed by an alpha sample.
   
   6       8,16        Each pixel is an R,G,B triple,
                       followed by an alpha sample.
```